pyomo/contrib/gdp_bounds/plugins/compute_bounds.py

```diff
@@ -13,15 +13,28 @@
 from pyomo.core.expr.current import identify_variables
 from pyomo.core.kernel.component_set import ComponentSet
 from pyomo.core.kernel.component_map import ComponentMap
-from pyomo.core import (Constraint, Objective, ConstraintList,
-                           TransformationFactory, maximize, minimize, value)
+from pyomo.core import (Constraint, Objective,
+                        TransformationFactory, maximize, minimize, value)
 from pyomo.opt import SolverFactory
 from pyomo.gdp.disjunct import Disjunct
 from pyomo.core.plugins.transform.hierarchy import Transformation
 from pyomo.opt import TerminationCondition as tc
-import textwrap
-from six import iterkeys, iteritems
-linear_degrees = set([0, 1])
+
+linear_degrees = {0, 1}
+
+
+def disjunctive_bounds(scope):
+    """Return all of the variable bounds defined at a disjunctive scope."""
+    possible_disjunct = scope
+    while possible_disjunct is not None:
+        try:
+            return possible_disjunct._disj_var_bounds
```

Comment:
@qtothec Is it possible (and not too much trouble) to do this without accessing private attributes?

Reply:
Well, I actually stick `_disj_var_bounds` onto the Disjunct object if it doesn't exist, so I guess I could rename it to `disj_var_bounds`?

Reply:
@qtothec Oh, I misunderstood what was going on there. Okay. That is perfectly reasonable. I assume it is removed from the disjunct before the algorithm exits?

Reply:
No, because it is intended to be used elsewhere as well (e.g. in GDPopt)